src/lib/resolveMigrationDocumentData.ts

```diff
@@ -0,0 +1,239 @@
+	image: MigrationImage,
+	migration: Migration,
+	withThumbnails?: boolean,
+): FilledImageFieldImage | undefined => {
+	const { id: master, ...thumbnails } =
+		image instanceof PrismicMigrationAsset ? { id: image } : image
+
+	const asset = migration._assets.get(master.config.id)?.asset
+	const maybeInitialField = master.originalField
+
+	if (asset) {
+		const parameters = (maybeInitialField?.url || asset.url).split("?")[1]
+		const url = `${asset.url.split("?")[0]}${parameters ? `?${parameters}` : ""}`
+		const dimensions: FilledImageFieldImage["dimensions"] = {
+			width: asset.width!,
+			height: asset.height!,
+		}
+		const edit: FilledImageFieldImage["edit"] =
+			maybeInitialField && "edit" in maybeInitialField
+				? maybeInitialField?.edit
+				: { x: 0, y: 0, zoom: 1, background: "transparent" }
+
+		// We give priority to the asset's specific alt text, then the image's general alt text
+		const alt = master.config.alt || asset.alt || null
+
+		const resolvedThumbnails: Record<string, FilledImageFieldImage> = {}
+		if (withThumbnails) {
+			for (const [name, thumbnail] of Object.entries(thumbnails)) {
+				const resolvedThumbnail = resolveMigrationImage(thumbnail, migration)
+				if (resolvedThumbnail) {
+					resolvedThumbnails[name] = resolvedThumbnail
+				}
+			}
+		}
+
+		return {
+			id: asset.id,
+			url,
+			dimensions,
+			edit,
+			alt: alt,
+			copyright: asset.credits || null,
+			...resolvedThumbnails,
+		}
+	}
+}
+
+/**
+ * Resolves a migration rich text image node to a regular rich text image node.
+ *
+ * @param rtImageNode - Migration rich text image node to resolve.
+ * @param migration - Migration instance.
+ *
+ * @returns Resolved rich text image node.
+ */
+export const resolveMigrationRTImageNode = async (
+	rtImageNode: MigrationRTImageNode,
+	migration: Migration,
+): Promise<RTImageNode | undefined> => {
+	const image = resolveMigrationImage(rtImageNode.id, migration)
+
+	if (image) {
+		const linkTo = (await resolveMigrationDocumentData(
+			rtImageNode.linkTo,
+			migration,
+		)) as LinkField
+
+		return {
+			...image,
+			type: RichTextNodeType.image,
+			linkTo: isFilled.link(linkTo) ? linkTo : undefined,
+		}
+	}
+}
+
+/**
+ * Resolves a migration link to media to a regular link to media field.
+ *
+ * @param linkToMedia - Migration link to media to resolve.
+ * @param migration - Migration instance.
+ *
+ * @returns Resolved link to media field.
+ */
+export const resolveMigrationLinkToMedia = (
+	linkToMedia: MigrationLinkToMedia,
+	migration: Migration,
+): LinkToMediaField<"filled"> | undefined => {
+	const asset = migration._assets.get(linkToMedia.id.config.id)?.asset
+
+	if (asset) {
+		return {
+			id: asset.id,
+			link_type: LinkType.Media,
+			name: asset.filename,
+			kind: asset.kind,
+			url: asset.url,
+			size: `${asset.size}`,
+			height: typeof asset.height === "number" ? `${asset.height}` : undefined,
+			width: typeof asset.width === "number" ? `${asset.width}` : undefined,
```

Comment:
💡 #idea: I could be wrong, but I think all of these properties are ignored. The Migration API docs say a `link_type: "Media"` link only needs an `id` to its asset.
```suggestion
```

Reply:
OK, that explains why the API doesn't honor image edits correctly I guess. I'll simplify that nonetheless for media links as they don't need that kind of information.

Reply:
Simplified!